pkg/compactor/block_upload.go

```diff
@@ -365,6 +409,69 @@ func (c *MultitenantCompactor) uploadMeta(ctx context.Context, logger log.Logger
 	return nil
 }
 
+func (c *MultitenantCompactor) validateBlock(ctx context.Context, blockID ulid.ULID,
+	userBkt objstore.Bucket, meta metadata.Meta) error {
+
+	// create a temporary directory to download the block
+	// on normal compactors this will be in the compartor's data directory, but in other cases (like some tests)
+	// the fallback will be the system's temp directory
+	tmpDir := os.TempDir()
+	if _, err := os.Stat(c.compactorCfg.DataDir); !os.IsNotExist(err) {
+		tmpDir = c.compactorCfg.DataDir
+	}
+	blockDir, err := os.MkdirTemp(tmpDir, "upload")
+
+	level.Debug(c.logger).Log("msg", "created temporary block directory", "dir", blockDir)
+	if err != nil {
+		return errors.Wrap(err, "failed to create temp dir")
+	}
+	defer func() {
+		level.Debug(c.logger).Log("msg", "removing temporary block directory", "dir", blockDir)
+		if err := os.RemoveAll(blockDir); err != nil {
+			level.Warn(c.logger).Log("msg", "failed to remove temp dir", "path", blockDir, "err", err)
+		}
+	}()
+
+	// download the block to local storage
+	level.Debug(c.logger).Log("msg", "downloading block from bucket", "block", blockID.String())
+	err = objstore.DownloadDir(ctx, c.logger, userBkt, blockID.String(), blockID.String(), blockDir)
+	if err != nil {
+		return errors.Wrap(err, "failed to download block")
+	}
+
+	// rename the temporary meta file name to the expected one locally so that the block can be inspected
+	_, err = os.Stat(filepath.Join(blockDir, uploadingMetaFilename))
+	if err == nil {
```

Comment:
Shouldn't this be checking for an error?
```suggestion
	if err != nil {
```